ui/src/app/shared/components/revision.tsx

```diff
@@ -1,9 +1,12 @@
 import * as React from 'react';
 import {revisionUrl} from './urls';
 
-export const Revision = ({repoUrl, revision, children}: {repoUrl: string; revision: string; children?: React.ReactNode}) => {
+export const Revision = ({repoUrl, revision, path, children}: {repoUrl: string; revision: string; path?: string; children?: React.ReactNode}) => {
     revision = revision || '';
-    const url = revisionUrl(repoUrl, revision);
+    let url = revisionUrl(repoUrl, revision);
+    if (path) {
+        url += '/' + path;
```

Comment:
Looks like https://bitbucket.org/michael_crenshaw/test/commits/826b672ecb8be2963928a1362711a55b5d307e88/test/test.yaml fails.

I think we should add a `forPath` param to `revisionUrl`. If `forPath === true`, then we should always use `src`, even for commit hashes.

Reply:
why adding `forPath`? either way `commits` is not good

Reply:
`commits` makes sense if you're linking directly to an individual commit hash without a path appended.

So if we're constructing a link for a hash _without_ intending to append a path, then it's fine to return a `commits` path. (It's actually preferable, because bitbucket will render a diff).

If we're constructing a link for a hash intending to append a path, then `revisionUrl` should opt to use `src` instead of `commits`, because `commits` does not work for a specific file path.

Reply:
If I understand right, this should be the url in case of `path` ?
https://bitbucket.org/Nirshtain2605/yamls/src/f76db049fcd005d0bec5a2a4d4f32b7b4e0995ca/staging/ (valid path)

Reply:
Yep! Gotta use `src` to make the path work.

Reply:
Update it, wdyt ?

Reply:
I like it! Gotta test, but I think this is it.